test-acceptance/src/org/omegat/gui/editor/EditorUtilsTest.java

```diff
@@ -0,0 +1,123 @@
+import org.junit.experimental.runners.Enclosed;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+
+import org.omegat.core.Core;
+import org.omegat.core.CoreEvents;
+import org.omegat.gui.main.ProjectUICommands;
+import org.omegat.gui.main.TestCoreGUI;
+import org.omegat.util.LocaleRule;
+import org.omegat.util.Preferences;
+
+/**
+ * @author Hiroshi Miura
+ */
+@RunWith(Enclosed.class)
+public class EditorUtilsTest {
+
+    public static class EditorUtilsFirstStepsTest extends TestCoreGUI {
+
+        @Rule
+        public final LocaleRule localeRule = new LocaleRule(new Locale("en"));
+
+        @Test
+        public void testEditorUtilsGetWordFirstSteps() throws BadLocationException {
+            int offs = 518;
+            JTextComponent editPane = window.panel("First Steps").textBox("IntroPane").target();
+            int posStart = EditorUtils.getWordStart(editPane, offs);
+            int posEnd = EditorUtils.getWordEnd(editPane, offs);
+            String word = editPane.getText(posStart, posEnd - posStart);
+            assertEquals("translation", word);
+            assertEquals(508, posStart);
+            assertEquals(519, posEnd);
+        }
+    }
+
+    public static class EditorUtilsLoadedProjectTest extends TestCoreGUI {
+
+        @Rule
+        public final LocaleRule localeRule = new LocaleRule(new Locale("en"));
+
+        @Rule
+        public final TemporaryFolder folder = TemporaryFolder.builder().assureDeletion().build();
+
+        @Test
+        public void testEditorUtilsGetWordLoadedProject() throws Exception {
+            // Prepare a sample project
+            File tmpDir = folder.newFolder("omegat-sample-project-");
+            File projSrc = new File("test-acceptance/data/project_CN_JP/");
+            FileUtils.copyDirectory(projSrc, tmpDir);
+            FileUtils.forceDeleteOnExit(tmpDir);
+            Preferences.setPreference(Preferences.PROJECT_FILES_SHOW_ON_LOAD, false);
+            // Load the project and wait a completion
+            CountDownLatch latch = new CountDownLatch(1);
+            CoreEvents.registerProjectChangeListener(eventType -> {
+                if (Core.getProject().isProjectLoaded()) {
+                    latch.countDown();
+                }
+            });
+            SwingUtilities.invokeAndWait(() -> ProjectUICommands.projectOpen(tmpDir, true));
+            try {
+                assertTrue(latch.await(5, TimeUnit.SECONDS));
+            } catch (InterruptedException ignored) {
+            }
+            //
+            final JTextComponent editPane = window.panel("Editor - source.txt").textBox().target();
+            // select word from a source text
+            int offs = 102;
+            int posStart = EditorUtils.getWordStart(editPane, offs);
+            int posEnd = EditorUtils.getWordEnd(editPane, offs);
+            String word = editPane.getText(posStart, posEnd - posStart);
+            assertEquals("太平寺中的文笔塔", word);
```

Comment:
Chinese does not work correctly. It is because java has a bug which marked as "Won't-Fix".
https://bugs.openjdk.org/browse/JDK-4095325

There is a comment in Stack-overflow about it, and suggest to use ICU4j for that.
https://stackoverflow.com/a/44508826